src/ui/castleinternaljoystickslinux.pas

```diff
@@ -145,54 +147,70 @@ procedure TLinuxJoysticksBackend.Poll(const List: TJoystickList;
   event : TLinuxJsEvent;
   Joystick: TJoystick;
   BackendInfo: TLinuxJoystickBackendInfo;
+  BytesRead: TSsize;
+  JoystickHadBeenDisconnected: Boolean;
```

Comment:
Rename to  `JoystickHasBeenDisconnected` (had been -> has been), I think this is more correct in English ("has been" is used to indicate something happened in the past, like "has been turned off"; "had been" indicates a longer task that started and finished in past,  like "had been learning"). https://www.quora.com/When-should-I-use-has-been-have-been-and-had-been-1?share=1 , https://www.quora.com/How-can-I-determine-when-to-use-had-have-had-been-etc , https://www.quora.com/Whats-the-difference-between-had-been-and-has-been .

I know, nitpicking :)